src/System.Security.Cryptography.Algorithms/tests/RSATests.cs

```diff
@@ -192,6 +193,28 @@ public void VerifyDataStream_UsesHashDataAndVerifyHash()
             Assert.True(rsa.VerifyData(new MemoryStream(new byte[] { 42 }), new byte[1] { 24 }, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1));
         }
 
+        [Fact]
+        public void RSAEncryptionPadding_Equality()
+        {
+            var paddingType = typeof(RSAEncryptionPadding);
+            var ctorTypes = new Type[] { typeof(RSAEncryptionPaddingMode), typeof(HashAlgorithmName) };
+            ConstructorInfo paddingCtor = paddingType.GetConstructor(BindingFlags.NonPublic | BindingFlags.Instance, null, ctorTypes, null);
```

Comment:
Reflecting over framework internals can fail on UAPAoT, but I don't think you need to for this case. Pkcs1 doesn't have an associated hash algorithm, but Oaep does, so you can work with RSAEncryptionPadding.Pkcs1 and call RSAEncryptionPadding.CreateOaep to pick different HashAlgorithmNames.